azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/NicIpConfigurationImpl.java

```diff
@@ -237,16 +239,22 @@ public NicIpConfigurationImpl withSubnet(String name) {
     }
 
     @Override
-    public NicIpConfigurationImpl withExistingLoadBalancer(LoadBalancer loadBalancer) {
-        this.loadBalancerToAssociate = loadBalancer;
-        return this;
+    public NicIpConfigurationImpl withExistingLoadBalancerBackend(LoadBalancer loadBalancer, String backendName) {
+        for (BackendAddressPoolInner pool : loadBalancer.inner().backendAddressPools()) {
+            if (pool.name().equalsIgnoreCase(backendName)) {
+                ensureBackendAddressPools().add(pool);
+                return this;
+            }
+        }
+
+        return null;
```

Comment:
ah ok. I do have similar situation in VMSS, there i just throw `ArgumentException` with a message so that user knows whats wrong, returning null seems fine.